olive/engine/engine.py

```diff
@@ -344,8 +402,23 @@ def run(
 
         for accelerator_spec in self.accelerator_specs:
             with self.create_managed_environment(accelerator_spec):
+                if self.enable_auto_optimizer:
```

Comment:
engine itself should be transparent about whether the pass is from auto-optimizer or registered by user explicitly. 

From this point of view, the engine just do register and run. prepare_passes_from_auto_optimizer should be splited into pass generation and pass registration. For pass generation, it should be outside this class's responsibility.

Reply:
Will refactor it in following PR.